.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,279 @@
+        # Check OS and python setup
+        echo "OS: ${OS_NAME}"
+        which python
+        which pip
+        pip list
+        conda info
+        conda list
+
+    - name: install_hole
+      if : matrix.install_hole
+      run: |
+        # We manually build hole2 to avoid OS incompatibilities
+        git clone https://github.com/MDAnalysis/hole2.git
+        cd hole2/src
+        source ../source.apache
+        (make FC=${FC}) && (make PREFIX=${HOME}/hole2 FC=${FC} install)
+        source ../source.unset
+        echo "HOLE_BINDIR=${PWD}/../exe" >> $GITHUB_ENV
+        echo "${PWD}/../exe" >> $GITHUB_PATH
+
+    - name: install_mda
+      run: |
+        (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
+
+    - name: run_tests
+      if: contains(matrix.name, 'asv_check') != true
+      run: |
+        PYTEST_FLAGS="--disable-pytest-warnings --durations=50"
+        if [ ${{ matrix.codecov }} = "true" ]; then
+          PYTEST_FLAGS="${PYTEST_FLAGS} --cov=MDAnalysis --cov-report=xml"
+        fi
+        echo $PYTEST_FLAGS
+        pytest -n 2 testsuite/MDAnalysisTests $PYTEST_FLAGS
+
+    - name: run_asv
+      if: contains(matrix.name, 'asv_check')
+      run: |
+        cd benchmarks
+        time python -m asv check -E existing
+
+    - name: codecov
+      if: matrix.codecov
+      uses: codecov/codecov-action@v1
+      with:
+        file: coverage.xml
+        fail_ci_if_error: True
+        verbose: True
+
+
+  build_docs:
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ubuntu-latest
+    env:
+      CYTHON_TRACE_NOGIL: 1
+      MPLBACKEND: agg
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: 3.7
+        auto-update-conda: true
+        channel-priority: flexible
+        channels: biobuilds, conda-forge
+        add-pip-as-python-dependency: true
+        architecture: x64
+
+    - name: install_deps
+      run: |
+        conda_deps="${{ env.MDA_CONDA_MIN_DEPS }} ${{ env.MDA_CONDA_EXTRA_DEPS}}"
+        pip_deps="${{ env.MDA_PIP_MIN_DEPS}} ${{ env.MDA_PIP_EXTRA_DEPS }} sphinx==1.8.5 sphinx-sitemap sphinx_rtd_theme msmb_theme==1.2.0"
+        conda install ${conda_deps}
+        pip install ${pip_deps}
+
+    - name: install_mda
+      run: |
+        cd package && python setup.py build_ext --inplace
+
+    - name: build_docs
+      run: |
+        cd package && python setup.py build_sphinx
+
+    - name: deploy_docs
+      if: github.event_name != 'pull_request'
+      run: |
+        # place the deploy call here
+        echo "Oh, maple syrup roast parsnips [Richard Gowers]"
+
+
+  pylint_check:
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: 3.6
```

Comment:
bump to 3.7?

Reply:
Done, I've also bumped up everything except from the minimum dep and the minimum numpy version builds. My reasoning for the latter two is that they are specifically around to test the "edge case with the least amount of features".